### PR TITLE
fix: clicking on disabled fast-button events should behave like native button (#4884)

### LINKED ISSUE #4884: fix: Click events on disabled fast-button bubble up the parent hierarchy
# 🐛 Bug Report

When a fast button is in disabled state click events on it are still propagated up the parent hierarchy.

This may be releated to #4698 and #4137, but since the observed behaviour does not affect the event listeners on the fast element, but the event propagation I am creating this separate issue.

The issue propably affects other input elements as well, but I only tested this on fast button for now.

## 💻 Repro or Code Sample

See [https://skaempfer.github.io/fast-button-click/](https://skaempfer.github.io/fast-button-click/) for a behaviour comparison between native button element and fast-button element 

## 🤔 Expected Behavior

Fast button element behaves like native button element: When in disabled state, fast-button elements do not propagate click events up their parent hierarchy. This should be in accordance with the HTML spec: "*A form control that is disabled must prevent any click events that are queued on the user interaction task source from being dispatched on the element.*" ([Source](https://www.w3.org/TR/html52/sec-forms.html#enabling-and-disabling-form-controls-the-disabled-attribute))

## 😯 Current Behavior

When you click on an disabled fast button any click listener on the element is not triggered, but the click event bubbles up the parent hierarchy.

## 💁 Possible Solution

The fast button element wraps a native button element. To behave like its native counterpart the fast button should make a descision on blocking or allowing event propagation based on the disabled state of the wrapped button element.

### PR DESCRIPTION
**Pull request by prudepixie**
# Pull Request

## 📖 Description
This change makes sure that the click events from fast-button in its disabled state behaves the same way as a native html button. 
Specifically, if text is not wrapped around addition html element, clicking on the button should not propagate events to the container div.
```
<div id="container">
   <fast-button disabled>Button</fast-button>
</div>
```

If in the following example, there is a span wrapped around text, then clicking on the span would propagate to the container as a span within a native button would behave.
```
<div id="container">
   <fast-button disabled><span>Button</span></fast-button>
</div>
```

### 🎫 Issues

fixes #4884 
### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by nicholasrice** [comment]
Thanks @prudepixie! can you change the issue reference to say "fixes #4884"? The "fixes" tells github to close the issue when the PR is merged.

**Review by nicholasrice — APPROVED** [review]

**Comment by EisenbergEffect** [comment]
Woohoo. 🎊 First official team member PR I think.

**Comment by EisenbergEffect** [comment]
I've reviewed and it looks good to me. I'd like to get @chrisdholt or a member of his team to take a look since they were addressing a few similar issues in other components recently.

**Comment by chrisdholt** [comment]
> I've reviewed and it looks good to me. I'd like to get @chrisdholt or a member of his team to take a look since they were addressing a few similar issues in other components recently.

@eljefe223 is on point to fix these across the components, so I’ve asked him to review this.

File: packages/web-components/fast-components/src/styles/patterns/button.ts [hunk 1]
```diff
@@ -92,6 +92,15 @@ export const BaseButtonStyles = css`
         border: 0;
     }
 
+    .control:disabled span
```
Comment (eljefe223) [inline packages/web-components/fast-components/src/styles/patterns/button.ts:95]: Thanks for the PR. Curious thought, this only applies to `span`?

**Review by eljefe223 — APPROVED** [review]

File: packages/web-components/fast-components/src/styles/patterns/button.ts (hunk 1 above)
Comment (prudepixie) [reply packages/web-components/fast-components/src/styles/patterns/button.ts:95]: The button template for the fast-button is defined as:
```
#shadow-root
<button>
  <span part='start></span>
  <span class='content'><slot></slot></span>
  <span part='end'></span>
</button>
```
so the styles are specific to the spans here

Comment (eljefe223) [reply packages/web-components/fast-components/src/styles/patterns/button.ts:95]: Ok, I think I misunderstood the problem you were trying to solve based on the description you provided.

**Comment by prudepixie** [comment]
A potentially more accurate solution for this issue: https://github.com/microsoft/fast/pull/5436

**Pull request closed without merging**